contracts/AgentMech.sol

```diff
@@ -156,45 +217,73 @@ contract AgentMech is ERC721Mech {
         // Increase the total number of requests
         numTotalRequests++;
 
-        emit Request(msg.sender, requestId, requestIdWithNonce, data);
+        emit Request(account, requestId, requestIdWithNonce, data);
     }
 
-    /// @dev Performs actions before the delivery of a request.
-    /// @param data Self-descriptive opaque data-blob.
-    /// @return requestData Data for the request processing.
-    function _preDeliver(uint256, bytes memory data) internal virtual returns (bytes memory requestData) {
-        requestData = data;
+    /// @dev Revokes the request from the mech that does not deliver it.
+    /// @notice Only marketplace can call this function if the request is not delivered by the chosen priority mech.
+    /// @param requestIdWithNonce Request Id with nonce.
+    function revokeRequest(uint256 requestIdWithNonce) external {
```

Comment:
When the request is delivered not by the priority mech, it must be revoked from the priority mech.